sympy/solvers/ode/single.py

```diff
@@ -651,3 +653,79 @@ def _get_general_solution(self, *, simplify: bool = True):
             ) * exp(-(1 - n)*Integral(P, x)))**(1/(1 - n))
         )
         return [gensol]
+
+
+class Factorable(SingleODESolver):
+
+    hint = "factorable"
+    has_integral = False
+
+    def _matches(self):
+        eq = self.ode_problem.eq
+        f = self.ode_problem.func.func
+        x = self.ode_problem.sym
+        order =self.ode_problem.order
+        df = f(x).diff(x)
+        eqs = []
+        eq = eq.collect(f(x), func = cancel)
+        eq = fraction(factor(eq))[0]
+        roots = factor_list(eq)[1]
+        if len(roots)>1 or roots[0][1]>1:
+            for i in roots:
+                if i[0].has(f(x)):
+                    eqs.append(i[0])
+            if len(eqs)>0:
+                self.check = {'eqns' : eqs}
+                return True
+        roots = solve(eq, df)
```

Comment:
Does this work for ODEs higher than first order?

Reply:
Yes, It works.